### PR TITLE
added Angular Typescript blogpost

### PR DESCRIPTION
**Pull request by KevinDaHub**

### CODE REVIEW

File: _posts/2016-03-16-AngularTS.md [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+tags: [TypeScript, AngularJS]
+category: AngularJS, TypeScript
+comments: true
+---
+
+## Combining the best of two worlds.
+
+Since my introduction to the heroic AngularJS framework at Devoxx around 4 years ago.
+I was intrigued and set for an adventure.
+With Angular 2 release in sight we have to prepare ourselves with the migrating road map coming up.
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Would reword to:
**With the upcoming release of Angular 2** we have ...

File: _posts/2016-03-16-AngularTS.md [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+One of the core changes in angular, was changing the code language to TypeScript.
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Would delete the comma.

File: _posts/2016-03-16-AngularTS.md [hunk 3]
```diff
@@ -0,0 +1,413 @@
...
+This blog will cover the use of angular components in TypeScript.
+But what is this TypeScript? 
+TypeScript is a superset of JavaScript that focuses on strong typing and the new ES6 features: classes, interfaces and modules.
+Like in common OO languages like Java and C# these features aren’t new.
+These features gives the developer the opportunity to build a object oriented architecture in JavaScript.
+With that in mind let’s see what the advantages are:
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Add a comma:
With that in mind**,** let's see...

File: _posts/2016-03-16-AngularTS.md [hunk 4]
```diff
@@ -0,0 +1,413 @@
...
+This blog will cover the use of angular components in TypeScript.
+But what is this TypeScript? 
+TypeScript is a superset of JavaScript that focuses on strong typing and the new ES6 features: classes, interfaces and modules.
+Like in common OO languages like Java and C# these features aren’t new.
+These features gives the developer the opportunity to build a object oriented architecture in JavaScript.
+With that in mind let’s see what the advantages are: 
+
+### Transpiling
+The DOM on your website can only recognize JavaScript.
+With this said they had to come with a way to compile TS.
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Would change to:
With this said**,** they had to come **up** with...

File: _posts/2016-03-16-AngularTS.md [hunk 5]
```diff
@@ -0,0 +1,413 @@
...
+Transpilers are integrated in the latest IDE’s. 
+Any valid JavaScript is valid TypeScript.
+
+### Strongly typed
+When you are used to plain JavaScript, you notice that every time you need a variable, your type is loosely typed.
+With TypeScript they give you the opportunity to give every variable its own type.
+This comes with great beneficials like better refactoring, less bugs and better type checking at compile time.
+
+### OO architecture
+TypeScript offers a object oriented architecture experience, which means all code is defined in classes, interfaces and most of those classes can be instantiated into objects. It also supports encapsulation, which protects the data from unintended access and modification.
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Change to:
... offers a**n** object ...

File: _posts/2016-03-16-AngularTS.md [hunk 6]
```diff
@@ -0,0 +1,413 @@
...
+### Strongly typed
+When you are used to plain JavaScript, you notice that every time you need a variable, your type is loosely typed.
+With TypeScript they give you the opportunity to give every variable its own type.
+This comes with great beneficials like better refactoring, less bugs and better type checking at compile time.
+
+### OO architecture
+TypeScript offers a object oriented architecture experience, which means all code is defined in classes, interfaces and most of those classes can be instantiated into objects. It also supports encapsulation, which protects the data from unintended access and modification. 
+
+
+## Learning path AngularTS
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Would change to:
Learning path **of** AngularTS

File: _posts/2016-03-16-AngularTS.md [hunk 7]
```diff
@@ -0,0 +1,413 @@
...
+If you’re no stranger to AngularJS you will notice that the structure remains the same. 
+Two way databinding, controllers, services, ... 
+But be aware that it has a different syntax in TypeScript. 
+I will show you the different best practices to implement these components. 
+
+
+## TypeScript Definition Files
+When using TS we have to reference to TSD files, these files describe the types defined in external libraries such as angular.
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Replace comma by period:
When using TS we have to reference to TSD files**. T**hese...

Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Capitalize Angular:
...such as **A**ngular.

File: _posts/2016-03-16-AngularTS.md [hunk 8]
```diff
@@ -0,0 +1,413 @@
...
+## Learning path AngularTS
+If you’re no stranger to AngularJS you will notice that the structure remains the same. 
+Two way databinding, controllers, services, ... 
+But be aware that it has a different syntax in TypeScript. 
+I will show you the different best practices to implement these components. 
+
+
+## TypeScript Definition Files
+When using TS we have to reference to TSD files, these files describe the types defined in external libraries such as angular. 
+To install the angular tsd files we use the tsd manager.
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Probably best to always capitalise "TSD"?
Is TSD manager still actual? According to the repo it's deprecated and replaced by Typings: https://github.com/Definitelytyped/tsd#install

File: _posts/2016-03-16-AngularTS.md [hunk 9]
```diff
@@ -0,0 +1,413 @@
...
+To use the tsd manager we install it with:
+{% highlight bash %}
+npm install tsd -g
+{% endhighlight %}
+
+Afterwards install angular with:
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Would put the explanation parts outside the code block and use backticks (\`) when referring to the options:
`--resolve` will include all dependencies angular needs.
`--save` will include the angular tsd into the tsd.json file.

File: _posts/2016-03-16-AngularTS.md [hunk 10]
```diff
@@ -0,0 +1,413 @@
...
+{% highlight bash %}
+tsd install angular --resolve --save 
+'--resolve' will include all dependencies angular needs.
+'--save' will include the angular tsd into the tsd.json file.
+{% endhighlight %}
+
+When installed you have a typings folder where you can find all your tsd files.
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Would style to:
... you have a `typings` folder ...

File: _posts/2016-03-16-AngularTS.md [hunk 11]
```diff
@@ -0,0 +1,413 @@
...
+{% highlight bash %}
+/// <reference path="../../typings/angularjs/angular.d.ts" />
+/// <reference path="../../typings/angularjs/angular-route.d.ts" />
+{% endhighlight %}
+
+## Modules
+
+### Angular Modules
+Modules are here to help us modularize our code.
+But it is important to know that if you’re not planning to share or declare related functionality.
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Don't really understand this sentence, maybe reword it a bit?

File: _posts/2016-03-16-AngularTS.md [hunk 12]
```diff
@@ -0,0 +1,413 @@
...
+It is best practice to use only one main module as the root of your application.
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Change to:
It is **a** best practice ...

File: _posts/2016-03-16-AngularTS.md [hunk 13]
```diff
@@ -0,0 +1,413 @@
...
+/// <reference path="../../typings/angularjs/angular-route.d.ts" />
+{% endhighlight %}
+
+## Modules
+
+### Angular Modules
+Modules are here to help us modularize our code.
+But it is important to know that if you’re not planning to share or declare related functionality.
+It is best practice to use only one main module as the root of your application. 
+To let the module know the existence of every component, they have to register themselves.
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Change to:
... let the module know **of** the existence of ...

File: _posts/2016-03-16-AngularTS.md [hunk 14]
```diff
@@ -0,0 +1,413 @@
...
+When registering the module you have to add all the libraries you want to depend upon.
+In this example we inject the routing service for navigation. 
+
+{% highlight javascript %}
+module JWorks {
+    "use strict";
+
+    angular
+       .module("jworks360", ['ngRoute'])
+     }
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Align the closing bracket.

File: _posts/2016-03-16-AngularTS.md [hunk 15]
```diff
@@ -0,0 +1,413 @@
...
+{% endhighlight %}
+
+### Internal TypeScript Modules
+These modules are similar to namespaces.
+You can define an unique namespace around your code.
+This will encapsulate variables, interfaces, and classes.
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Remove the last comma:
..., interfaces and classes.

File: _posts/2016-03-16-AngularTS.md [hunk 16]
```diff
@@ -0,0 +1,413 @@
...
+#### Transpiled JavaScript
+
+{% highlight javascript %}
+var JWorks;
+(function(JWorks){
+
+})(JWorks || (JWorks = {}));
+{% endhighlight %}
+
+To encapsulate our code, the module will transpile to an IIFE around our components.
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Would add between parentheses what the IFFE abbreviation stands for.

File: _posts/2016-03-16-AngularTS.md [hunk 17]
```diff
@@ -0,0 +1,413 @@
...
+This will avoid global code which helps prevent variables and function declarations from living longer than expected in the global scope, which also helps avoid variable collisions.
+
+
+## Entity Class
+Now that TypeScript supports OO, we can analyse our business problem and define the business objects into entity classes.
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Would write "OO" as "Object-oriented programming".

File: _posts/2016-03-16-AngularTS.md [hunk 18]
```diff
@@ -0,0 +1,413 @@
...
+When you analyse and define these entities you can define which properties and methods each entity needs.
+If you have a couple entities, you can even establish a relationship.
+This will provide a clear view on what you want to achieve and have the possibility to create multiple instances of these classes. 
+When building an entity class you can optionally define an interface to show what’s the intent.
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Would reword to:
... to show what the intention of the class is.

File: _posts/2016-03-16-AngularTS.md [hunk 19]
```diff
@@ -0,0 +1,413 @@
...
+
+
+## Entity Class
+Now that TypeScript supports OO, we can analyse our business problem and define the business objects into entity classes.
+When you analyse and define these entities you can define which properties and methods each entity needs.
+If you have a couple entities, you can even establish a relationship.
+This will provide a clear view on what you want to achieve and have the possibility to create multiple instances of these classes. 
+When building an entity class you can optionally define an interface to show what’s the intent. 
+
+{% highlight javascript %}
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Isn't there a TypeScript highlight available? Looks a bit odd with the colons marked in red.

File: _posts/2016-03-16-AngularTS.md [hunk 20]
```diff
@@ -0,0 +1,413 @@
...
+
+{% endhighlight %}
+
+To use your entity class in a controller you have to define the export key.
+This will expose the class to other classes.
+When exporting the interface you will use it as a data type.
+
+### Class as property
+{% highlight javascript %}
+Employee :IEmployee;
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Remove the space before the colon in order to be consistent with the other code samples?

File: _posts/2016-03-16-AngularTS.md [hunk 21]
```diff
@@ -0,0 +1,413 @@
...
+    interface IEmployeeController{
+        person:IPerson;
+        save(person:IPerson):void;
+    }
+}
+{% endhighlight %}
+
+The interface will show you the intent of our controller and declare the properties and methods that will be used.
+When you look at the syntax, you see that the properties are strong typed and the type is declared after the colon.
+If you have no aware of which type a property has, you can choose for the general type ‘any’.
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Change to:
If you **aren't aware of the type of a property, you can fall back to** the general type `any`.

File: _posts/2016-03-16-AngularTS.md [hunk 22]
```diff
@@ -0,0 +1,413 @@
...
+For the methods declared in the interface you have to give the needed parameters and the return type.
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Change to:
... you have to **specify the required** parameters ...

File: _posts/2016-03-16-AngularTS.md [hunk 23]
```diff
@@ -0,0 +1,413 @@
...
+
+{% highlight javascript %}
+module JWorks {
+
+    class EmployeeController implements IEmployeeController{
+
+        static $inject = ["EmployeeService"];
+
+        constructor(private employeeService:IEmployeeService,public employee:IEmployee){
+             this.employee = new Employee(“Nivek”,”Kevin”);
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Double quotes seem off? Are they "MS Word"-quotes?
Probably need to be replaced by actual double quotes (") used to define a String.

File: _posts/2016-03-16-AngularTS.md [hunk 24]
```diff
@@ -0,0 +1,413 @@
...
+        });
+   }
+
+}
+angular.module("jworks360")
+.controller("EmployeeController", EmployeeController);
+{% endhighlight %}
+
+### Dependency Injection in classes
+When a service is needed in your controller we have to inject these in order to use them.
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Change to:
... needed in your controller**, it needs to be injected in order to us it.**

File: _posts/2016-03-16-AngularTS.md [hunk 25]
```diff
@@ -0,0 +1,413 @@
...
+In the above example it is important you declare the ‘static $injection’ above your constructor.
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Use backsticks (\`) to refer to code snippets:
`static $injection`

File: _posts/2016-03-16-AngularTS.md [hunk 26]
```diff
@@ -0,0 +1,413 @@
...
+
+{% highlight bash %}
+/// <reference path="../services/employee.service.ts"/>
+{% endhighlight %}
+
+
+### Constructor
+TypeScript supports initialisation of your properties and injections in a constructor.
+When declaring properties in your class, you can declare them directly into your constructor.
+As these two examples are right you can have issues in your tests with the second example.
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Would reword to:
**Although **these two examples are **correct** you can have ...

File: _posts/2016-03-16-AngularTS.md [hunk 27]
```diff
@@ -0,0 +1,413 @@
...
+    }
+
+}
+//Becomes:
+class Controller {
+
+    constructor(public name:string){
+    }
+
+
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Missing a closing bracket.

File: _posts/2016-03-16-AngularTS.md [hunk 28]
```diff
@@ -0,0 +1,413 @@
...
+{% endhighlight %}
+Be sure to notify that we are using access modifiers to tell the controller which properties we want to expose to the view.
+The best practice is that you put your injections and angular services private and all your properties you want to use on your view public.
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Capitalise Angular?

File: _posts/2016-03-16-AngularTS.md [hunk 29]
```diff
@@ -0,0 +1,413 @@
...
+When initialising strings, TypeScript makes no differ between double or single quotes.
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Change to:
... makes no **distinction** between ...

File: _posts/2016-03-16-AngularTS.md [hunk 30]
```diff
@@ -0,0 +1,413 @@
...
+
+### ControllerAs
+
+Controller classes uses the controllerAs feature as default.
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: 'Use' instead of 'uses'.

File: _posts/2016-03-16-AngularTS.md [hunk 31]
```diff
@@ -0,0 +1,413 @@
...
+   }
+   routes.$inject = ["$routeProvider"];
+
+   angular.module("jworks360")
+       .config(routes);
+}
+{% endhighlight %}
+
+## Services
+When you make a custom service, the code you implement is reusable and can be called in any other angular components including controllers and other services.
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Capitalise Angular?

File: _posts/2016-03-16-AngularTS.md [hunk 32]
```diff
@@ -0,0 +1,413 @@
...
+It is important to know that services are singletons, so there will be only one instance of the used service.
+With this in mind we can use the custom service to share data across all components in angular for example communicating with a http service to collect data and share it with any other component by injecting the service.
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Capitalise Angular?

Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Capitalise 'HTTP'.

File: _posts/2016-03-16-AngularTS.md [hunk 33]
```diff
@@ -0,0 +1,413 @@
...
+
+### RestAngular
+For my project I used an Angular service that simplifies common verb requests with a minimum of client code.
+In my custom services you will see examples of restangular in TypeScript.
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Properly capitalise RestAngular.

File: _posts/2016-03-16-AngularTS.md [hunk 34]
```diff
@@ -0,0 +1,413 @@
...
+If you like to checkout what the difference is with $resource, you can check this [list](https://github.com/mgonto/restangular#differences-with-resource)
+
+{% highlight javascript %}
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Reformat the code because it seems a bit off.

File: _posts/2016-03-16-AngularTS.md [hunk 35]
```diff
@@ -0,0 +1,413 @@
...
+              getEmployeeByLink(href:string):ng.IPromise<{}>{
+                  return this.employees.oneUrl(href).get();
+              }       
+          }
+          angular.module("jworks360")
+              .service("EmployeeService", EmployeeService);       
+       }
+{% endhighlight %}
+
+In the above example, restangular has its own tsd file to reference to.
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: RestAngular.

Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Capitalise TSD?

File: _posts/2016-03-16-AngularTS.md [hunk 36]
```diff
@@ -0,0 +1,413 @@
...
+For services it is best practice to declare an interface for data typing and get a clear view of the intent.
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Change to:
... it is **a** best practice ...
... data typing and **getting** a clear view ...

File: _posts/2016-03-16-AngularTS.md [hunk 37]
```diff
@@ -0,0 +1,413 @@
...
+The service class will implement all methods related to the data communication with the backend and returns a promise to the controllers or services that will inject this custom service.
+RestAngular has it’s own configuration you can modify in the .config component to point to the right api call.
+After the config you can inject the restangular service and use its services to build up a request to the backend.
+
+
+## Directives
+Custom directives allow you to create highly semantic and reusable components.
+A directive allows angular to manipulate the DOM and add it’s own behaviours.
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: "Its" instead of "it's"

File: _posts/2016-03-16-AngularTS.md [hunk 38]
```diff
@@ -0,0 +1,413 @@
...
+The service class will implement all methods related to the data communication with the backend and returns a promise to the controllers or services that will inject this custom service.
+RestAngular has it’s own configuration you can modify in the .config component to point to the right api call.
+After the config you can inject the restangular service and use its services to build up a request to the backend.
+
+
+## Directives
+Custom directives allow you to create highly semantic and reusable components.
+A directive allows angular to manipulate the DOM and add it’s own behaviours. 
+These can be either a set of instructions or a JSON representation.
+Defining a directive in TypeScript we use the directive service angular provides.
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Reword to:
**To define** a directive in TypeScript we use the directive service** that Angular** provides.

File: _posts/2016-03-16-AngularTS.md [hunk 39]
```diff
@@ -0,0 +1,413 @@
...
+
+       }
+
+   }
+   angular.module("jworks360").directive("jwAnimate", Animate.instance);
+
+}
+{% endhighlight %}
+
+In the interface above we have to tell angular what name we will use for our directive.
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Capitalise Angular.

File: _posts/2016-03-16-AngularTS.md [hunk 40]
```diff
@@ -0,0 +1,413 @@
...
+The attribute service will be called to add an additional string to the attribute service.
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: The attribute service will add something to itself?

File: _posts/2016-03-16-AngularTS.md [hunk 41]
```diff
@@ -0,0 +1,413 @@
...
+Secondly the class has to implement the directive interface to be recognized by the compiler as a directive.
+Inside the class you have to declare the prefixed properties and override the methods you will be using.
+The ‘static instance()’ method has to be declared to let your module know that there is a new directive.
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Use backticks for the code snippet:
`static instance()`

File: _posts/2016-03-16-AngularTS.md [hunk 42]
```diff
@@ -0,0 +1,413 @@
...
+At the end you register the directive to your module with the instance as value.
+
+
+
+## Final note
+
+Best practices can change over time. 
+With webpack for example the registry to the module is gathered in one file.
+TypeScript keeps on growing and in my opinion will be the default language for many future front-end projects.
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Change to:
... and**,** in my opinion**, **will be ...

File: _posts/2016-03-16-AngularTS.md [hunk 43]
```diff
@@ -0,0 +1,413 @@
...
+When we arrive at testing, TypeScript provides better support because of encapsulation.
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Reword to:
When **it comes to testing our code**, TypeScript **will provide** better ...

File: _posts/2016-03-16-AngularTS.md [hunk 44]
```diff
@@ -0,0 +1,413 @@
...
+
+
+
+## Final note
+
+Best practices can change over time. 
+With webpack for example the registry to the module is gathered in one file.
+TypeScript keeps on growing and in my opinion will be the default language for many future front-end projects.
+When we arrive at testing, TypeScript provides better support because of encapsulation. 
+Finally, this is a nice learning road to take if you want to migrate to Angular 2.
```
Comment (yannickdeturck) [inline _posts/2016-03-16-AngularTS.md]: Maybe change to "Learning path"?

File: _posts/2016-03-16-AngularTS.md (hunk 32 above)
Comment (TYsewyn) [inline _posts/2016-03-16-AngularTS.md]: It should be 'an HTTP service'

File: _posts/2016-03-16-AngularTS.md [hunk 45]
```diff
@@ -0,0 +1,413 @@
...
+{% highlight bash %}
+tsd install angular --resolve --save 
+'--resolve' will include all dependencies angular needs.
+'--save' will include the angular tsd into the tsd.json file.
+{% endhighlight %}
+
+When installed you have a typings folder where you can find all your tsd files. 
+In the tsd.d.ts file you will see the references the application will use for angular. 
+It is possible that you have to declare the referencing on the top of your file.
+{% highlight bash %}
```
Comment (TYsewyn) [inline _posts/2016-03-16-AngularTS.md]: It is better to use type text.
If the implementation of the bash highlighter changes this could result in an incorrect formatted output.

File: _posts/2016-03-16-AngularTS.md [hunk 46]
```diff
@@ -0,0 +1,413 @@
...
+{% endhighlight %}
+
+### Dependency Injection in classes
+When a service is needed in your controller we have to inject these in order to use them.
+In the above example it is important you declare the ‘static $injection’ above your constructor.
+The reason behind this is that in your constructor you will initialise the injected services.
+By doing this the constructor will recognize the injection. 
+If you inject a custom service you have to reference to the related service. 
+
+{% highlight bash %}
```
Comment (TYsewyn) [inline _posts/2016-03-16-AngularTS.md]: It is better to use type text.
If the implementation of the bash highlighter changes this could result in an incorrect formatted output.

File: _posts/2016-03-16-AngularTS.md [hunk 47]
```diff
@@ -0,0 +1,413 @@
...
+## Combining the best of two worlds.
+
+Since my introduction to the heroic AngularJS framework at Devoxx around 4 years ago.
+I was intrigued and set for an adventure.
+With Angular 2 release in sight we have to prepare ourselves with the migrating road map coming up.
+One of the core changes in angular, was changing the code language to TypeScript.
+This blog will cover the use of angular components in TypeScript.
+But what is this TypeScript? 
+TypeScript is a superset of JavaScript that focuses on strong typing and the new ES6 features: classes, interfaces and modules.
+Like in common OO languages like Java and C# these features aren’t new.
```
Comment (MrJean) [inline _posts/2016-03-16-AngularTS.md]: Remove the first Like (In common ...) or "Like in common OO languages such as Java ..."

File: _posts/2016-03-16-AngularTS.md (hunk 3 above)
Comment (MrJean) [inline _posts/2016-03-16-AngularTS.md]: Remove colon? Since the next part is a heading and not a list.

File: _posts/2016-03-16-AngularTS.md (hunk 4 above)
Comment (MrJean) [inline _posts/2016-03-16-AngularTS.md]: With that said. Instead of With this said.

File: _posts/2016-03-16-AngularTS.md [hunk 48]
```diff
@@ -0,0 +1,413 @@
...
+These features gives the developer the opportunity to build a object oriented architecture in JavaScript.
+With that in mind let’s see what the advantages are: 
+
+### Transpiling
+The DOM on your website can only recognize JavaScript.
+With this said they had to come with a way to compile TS.
+Because TS is a superset of JS it can transpile to plain JavaScript before including it into HTML.
```
Comment (MrJean) [inline _posts/2016-03-16-AngularTS.md]: TypeScript is a compile. So change transpile to compile (you mentioned this earlier in the blog post).

File: _posts/2016-03-16-AngularTS.md [hunk 49]
```diff
@@ -0,0 +1,413 @@
...
+Transpilers are integrated in the latest IDE’s.
```
Comment (MrJean) [inline _posts/2016-03-16-AngularTS.md]: Transpilers <=> Compilers

File: _posts/2016-03-16-AngularTS.md [hunk 50]
```diff
@@ -0,0 +1,413 @@
...
+#### Transpiled JavaScript
+
+{% highlight javascript %}
+var JWorks;
+(function(JWorks){
+
+})(JWorks || (JWorks = {}));
+{% endhighlight %}
+
+To encapsulate our code, the module will transpile to an IIFE(Immediately-Invoked Function Expression) around our components.
```
Comment (MrJean) [inline _posts/2016-03-16-AngularTS.md]: Hate to say it, but maybe add a space between abbreviation and explanation? "IIFE (Ime ..."
